tests/integration/integration_util_test.go

```diff
@@ -649,3 +650,243 @@ func testConstructFailures(t *testing.T, lang string, dependencies ...string) {
+		if lang == "yaml" {
+			return nil
+		}
+		overrides := map[string]string{}
+		e := ptesting.NewEnvironment(t)
+		e.RunCommand("pulumi", "package", "gen-sdk", testProviderPath,
+			"--language", lang,
+			"--version", "1.0.0")
+		testProviderSDKPath, err := filepath.Abs(filepath.Join(e.RootPath, "sdk", lang))
+		require.NoErrorf(t, err, "filepath.Abs failed")
+		switch lang {
+		case "nodejs":
+			subE := *e
+			subE.CWD = testProviderSDKPath
+			subE.RunCommand("yarn", "install")
+			subE.RunCommand("yarn", "run", "build")
+
+			bin := filepath.Join(testProviderSDKPath, "bin")
+
+			bytes, err := os.ReadFile(filepath.Join(testProviderSDKPath, "package.json"))
+			require.NoErrorf(t, err, "package.json read failed")
+
+			err = os.WriteFile(filepath.Join(bin, "package.json"), bytes, 0600)
+			require.NoErrorf(t, err, "package.json copy failed")
+
+			overrides["@pulumi/testconfigprovider"] = fmt.Sprintf("file:%s", bin)
+		default:
+			t.Fatalf("Language not supported yet: %v", lang)
+		}
+		t.Logf("Generated test provider SDK to %s", testProviderSDKPath)
+		return overrides
+	})()
+
+	genPulumiYAML := func(t *testing.T, tc testCase) []byte {
+		program, err := yaml.Marshal(map[string]any{
+			"name":    "proj",
+			"runtime": "yaml",
+			"resources": map[string]any{
+				"provider": map[string]any{
+					"type": "pulumi:providers:testconfigprovider",
+					"properties": map[string]any{
+						tc.prop: tc.config,
+					},
+				},
+			},
+			"outputs": map[string]any{
+				"result": map[string]any{
+					"fn::invoke": map[string]any{
+						"function": "testconfigprovider:index:getConfig",
+						"options": map[string]any{
+							"provider": "${provider}",
+						},
+						"return": "result",
+					},
+				},
+			},
+			"plugins": map[string]any{
+				"providers": []any{
+					map[string]any{
+						"name": "testconfigprovider",
+						"path": testProviderPath,
+					},
+				},
+			},
+		})
+		require.NoError(t, err)
+		return program
+	}
+
+	genSourceDir := func(t *testing.T, tc testCase) string {
+		yamlProgram := genPulumiYAML(t, tc)
+		t.Logf("Testing YAML program:\n%s\n\n", string(yamlProgram))
+		switch lang {
+		case "yaml":
+			d := t.TempDir()
+			err := os.WriteFile(filepath.Join(d, "Pulumi.yaml"), yamlProgram, 0600)
+			require.NoError(t, err)
+			return d
+		default:
+			e := ptesting.NewEnvironment(t)
+			err := os.WriteFile(filepath.Join(e.RootPath, "Pulumi.yaml"), yamlProgram, 0600)
+			require.NoError(t, err)
+			e.RunCommand("pulumi", "convert", "--from", "yaml",
+				"--strict", "--non-interactive", "--generate-only",
+				"--language", lang, "--out", "out")
+			d := filepath.Join(e.RootPath, "out")
+			if lang == "nodejs" {
+				indexTsBytes, err := os.ReadFile(filepath.Join(d, "index.ts"))
+				require.NoErrorf(t, err, "failed to read index.ts")
+				t.Logf("Testing TypeScript program:\n%s\n\n", string(indexTsBytes))
+
+				indexTsBytes = bytes.ReplaceAll(indexTsBytes,
+					[]byte(`getConfigOutput({})`),
```

Comment:
There's a bug in `pulumi-convert` here somewhere, that skips sending resource option `provider` for an invoke when translating that from YAML to TS.

Reply:
https://github.com/pulumi/pulumi/issues/17525